### DISCUSSION TITLE
v13+ - enable or disable (or show/hide) a block in a blocklist based on ContentType

### DISCUSSION THREAD

**Discussion by afabri73** [umbraco/Umbraco-CMS#16312] [Features and ideas]
Hi, I use blocklists a lot and often reuse the same blocklist for different purposes.
It would be very useful, when configuring a block in a blocklist, to be able (optional) to specify which documentTypes can useit (multiple choice) or which documentTypes cannot use it (probably better).

This is an example:
I create a DocumentType Homepage with a blocklist that has two blocks AccordionBlock and SliderBlock
I create also a second DocumentType Article with a blocklist that should have only the block AccordionBlock.
At present, I must create two different blocklists, but it would be very useful if within the configuration of the SliderBlock I could select which DocumentTypes can use (Homepage) or cannot use (Article) the block.

This way, when I create a new article in content I couldn't add the slider

Have a nice day
Adriano

**Comment by hfloyd** [comment]
You can easily create a copy of a BlockList datatype and change what is available on it. Since Content Types each have their own properties, it would be possible to use one or the other datatypes on the different Content Types. 

This would be useful in the case of a Composition - if, for instance, you have a generic "Main Content" composition which you want to add to multiple page types, and your Block List property is on there... 

But it feels like a lot of extra work for the back-office to check the current Content Type before rendering the available blocks, however...

**Reply by afabri73** [reply]
Hi Heather,
I know I can easily copy the blocklist, but my goal was to minimize the number of blocklists to manage.
You're right that it might be too heavy for the backoffice to check Content Type before rendering the blocks, so I think the only way is to copy and manage a new blocklist, as you suggested.

Thank you again and have a nice day
Adriano

**Reply by jdpnielsen** [reply]
Being able to define rules for blocks would be great! I currently have to manage 4 instances of a block list in order to enable/disable certain blocks from certain document types. This is a big hassle when adding/changing blocks as i have to do this manually 4 times for each change.

If a concept of rules were added, i would also like to see something along the lines of: 
* user access rule (ie certain editors may only use/edit certain block types)
* max instance amount (ie. only one hero block per block list)

**Reply by nathanwoulfe** [reply]
Shameless self-promotion... Have a look at https://github.com/nathanwoulfe/Clip

It might do what you need, I honestly can't remember if I made it work with elements or just content items. If I didn't, it's probably possible with a bit of fiddling.

**Reply by jdpnielsen** [reply]
Interesting, @nathanwoulfe. I'll check it out! But i definitely think this is still worth considering as a Core feature.

**Comment by afabri73** [comment]
> Shameless self-promotion... Have a look at https://github.com/nathanwoulfe/Clip
> 
> It might do what you need, I honestly can't remember if I made it work with elements or just content items. If I didn't, it's probably possible with a bit of fiddling.

Very interesting @nathanwoulfe...thank you.
I'll try your code to test if this is what I'm searching.

For your info...on our.umbraco I also received another tip. Jamie Townsend suggested me to try to use the SendingContentNotification  (documentation: https://docs.umbraco.com/umbraco-cms/reference/notifications/editormodel-notifications)
I'll try also this way

Adriano
